sklearn/tests/test_discriminant_analysis.py

```diff
@@ -148,6 +148,18 @@ def test_lda_transform():
     assert_raise_message(NotImplementedError, msg, clf.transform, X)
 
 
+def test_lda_n_components():
+    clf = LinearDiscriminantAnalysis(solver="svd", n_components=2)
+    assert_warns_message(UserWarning, "'n_components' is not less than the "
+                         "number of classes, setting '_max_components' to "
+                         "'n_classes' - 1", clf.fit, X, y)
+    X_transformed = clf.fit(X, y).transform(X)
+    assert_equal(X_transformed.shape[1], 1)
```

Comment:
With the adoption of pytest, we are phasing out use of test helpers `assert_equal`, `assert_true`, etc. Please use bare `assert` statements, e.g. `assert x == y`, `assert not x`, etc.